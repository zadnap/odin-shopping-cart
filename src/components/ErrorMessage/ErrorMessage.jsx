import styles from './ErrorMessage.module.scss';
import errorFace from '@/assets/error-face.png';

function ErrorMessage({ message }) {
  return (
    <div role="alert" className={styles.errorMessage}>
      <img className={styles.errorImage} src={errorFace} alt="" />
      <p className={styles.message}>{message}</p>
    </div>
  );
}

export default ErrorMessage;
