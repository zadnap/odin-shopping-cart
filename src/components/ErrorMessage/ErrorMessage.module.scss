@use '@/styles/variables.scss' as *;

.errorMessage {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 15px;
  height: 100%;
  padding: 30px;
  background-color: $subbackground-color;
  border-radius: $border-radius;

  .errorImage {
    width: 70px;
  }
}
