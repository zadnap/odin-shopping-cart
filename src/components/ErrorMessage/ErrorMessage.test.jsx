import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import ErrorMessage from './ErrorMessage';

describe('ErrorMessage component', () => {
  it('should render to match snapshot', () => {
    const { container } = render(<ErrorMessage />);

    expect(container).toMatchSnapshot();
  });
});
