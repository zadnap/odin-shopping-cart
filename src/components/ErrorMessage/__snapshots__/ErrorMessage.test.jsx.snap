// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorMessage component > should render to match snapshot 1`] = `
<div>
  <div
    class="_errorMessage_2e14f5"
    role="alert"
  >
    <img
      alt=""
      class="_errorImage_2e14f5"
      src="/src/assets/error-face.png"
    />
    <p
      class="_message_2e14f5"
    />
  </div>
</div>
`;
